src/video/windows/SDL_windowsevents.c

```diff
@@ -2424,6 +2428,22 @@ LRESULT CALLBACK WIN_WindowProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lPara
 #endif // !defined(SDL_PLATFORM_XBOXONE) && !defined(SDL_PLATFORM_XBOXSERIES)
     }
 
+#ifdef HAVE_SHOBJIDL_CORE_H
+    if (data->videodata->WM_TASKBAR_BUTTON_CREATED) {
+        if (!data->videodata->taskbarlist) {
```

Comment:
Can we defer this until the progress bar APIs are used? It seems like we don't need to do this for all windows up front.

Reply:
Sure. While I was making this change, I also spotted a mistake: the condition to check if the WM_TASKBAR_BUTTON_CREATED was wrong, I forgot to compare it with msg. It's fixed.